### PR TITLE
Fix matrix expander

### PR DESCRIPTION
**Pull request by pablolh**
Let's fix modules one by one. This commit was initially only intended to move this piece of code to its own folder, to add structure to the codebase, but it turned out I discovered a bug in it.

- Fix untested bug where expanded matrix was wrong (not even unitary anymore)
- Move matrix expander to a utils folder
- Fix formatting
- Add auxiliary functions
- Add doctest and test

### CODE REVIEW

**Comment by pablolh** [comment]
This is the new branch with a new targeted issue linked

File: opensquirrel/utils/matrix_expander.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        10
+    """
+    result = 0
+    for i, bit_index in enumerate(bit_indices):
+        result |= ((x & (1 << i)) >> i) << bit_index
+
+    return result
+
+
+def clear_bits(x: int, bit_indices: list[int]) -> int:
```
Comment (elenbaasc) [inline opensquirrel/utils/matrix_expander.py:76]:
```suggestion
def clear_bits(x: int, bit_indices: List[int]) -> int:
```

File: opensquirrel/utils/matrix_expander.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        1
+    """
+    result = 0
+    for i, bit_index in enumerate(bit_indices):
+        result |= ((x & (1 << bit_index)) >> bit_index) << i
+
+    return result
+
+
+def deposit_bits(x: int, bit_indices: list[int]) -> int:
```
Comment (elenbaasc) [inline opensquirrel/utils/matrix_expander.py:42]:
```suggestion
def deposit_bits(x: int, bit_indices: List[int]) -> int:
```

File: opensquirrel/utils/matrix_expander.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        5
+    """
+    result = x
+    for index in bit_indices:
+        result &= ~(1 << index)
+
+    return result
+
+
+def get_expanded_matrix(semantic: Semantic, qubit_operands: list[int], total_qubits: int) -> np.ndarray:
```
Comment (elenbaasc) [inline opensquirrel/utils/matrix_expander.py:98]:
```suggestion
def get_expanded_matrix(semantic: Semantic, qubit_operands: List[int], total_qubits: int) -> np.ndarray:
```

**Pull request merged**